6-AdvancedScenarios/1-call-api-obo/SPA/src/components/DataDisplay.jsx

```diff
@@ -1,33 +1,84 @@
-import "../styles/App.css";
-import { protectedResources } from "../authConfig";
+import { createClaimsTable } from '../utils/claimUtils';
+import { Table } from 'react-bootstrap';
+import { protectedResources } from '../authConfig';
 
-export const HelloData = (props) => {
+export const IdTokenData = (props) => {
+    const tokenClaims = createClaimsTable(props.idTokenClaims);
+    const tableRow = Object.keys(tokenClaims).map((key) => {
+        return (
+            <tr key={key}>
+                {tokenClaims[key].map((claimItem) => (
+                    <td key={claimItem} className="tableColumn">
+                        {claimItem}
+                    </td>
+                ))}
+            </tr>
+        );
+    });
+    return (
+        <>
+            <div className="data-area-div">
+                <p>
+                    See below the claims in your <strong> ID token </strong>. For more information, visit:{' '}
+                    <span>
+                        <a href="https://docs.microsoft.com/en-us/azure/active-directory/develop/id-tokens#claims-in-an-id-token">
+                            docs.microsoft.com
+                        </a>
+                    </span>
+                </p>
+                <div className="data-area-div table">
+                    <Table responsive striped bordered hover>
+                        <thead>
+                            <tr>
+                                <th>Claim</th>
+                                <th>Value</th>
+                                <th>Description</th>
+                            </tr>
+                        </thead>
+                        <tbody>{tableRow}</tbody>
+                    </Table>
+                </div>
+            </div>
+        </>
+    );
+};
+
+export const ProfileData = (props) => {
     const tableRows = Object.entries(props.helloData).map((entry, index) => {
-        return (<tr key={index}>
-            <td><b>{entry[0]}: </b></td>
-            <td>{entry[1]}</td>
-        </tr>)
+        return (
+            <tr key={index}>
+                <td>
+                    <b>{entry[0]}: </b>
+                </td>
+                <td>{entry[1]}</td>
+            </tr>
+        );
```

Comment:
Consider using array destructuring here for readability:
```
    const tableRows = Object.entries(props.helloData).map(([ key, value ], index) => {
        return (
            <tr key={index}>
                <td>
                    <b>{key}: </b>
                </td>
                <td>{value}</td>
            </tr>
```

Reply:
Updated!